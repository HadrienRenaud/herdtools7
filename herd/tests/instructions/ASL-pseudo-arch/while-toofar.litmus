ASL while-toofar

{
x = 0;
0:X1=x;
}

func T0(x::bits(32))
begin
  var s: integer = 0;
  var i = UInt(read_memory(x, 32));
  let j = i;
  while i > 0 do
    s = s + i;
    i = i-1;
  end;
end;

func T1(x::bits(32))
begin
  for k = 1 to 3 do
    write_memory(x,32,k[2:0]);
  end;
end;

func main() => integer
begin
  let x = read_register(1);

  T0(x);
  T1(x);

  return 0;
end;

locations [0:T0.0.s;]
