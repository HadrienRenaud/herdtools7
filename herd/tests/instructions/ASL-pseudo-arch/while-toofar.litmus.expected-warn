Warning: File "./herd/tests/instructions/ASL-pseudo-arch/while-toofar.litmus": unrolling limit exceeded at while loop at file "./herd/tests/instructions/ASL-pseudo-arch/while-toofar.litmus", line 13 to line 16 pruned, legal outcomes may be missing.
