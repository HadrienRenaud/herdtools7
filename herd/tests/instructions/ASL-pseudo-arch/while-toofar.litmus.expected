Test while-toofar Required
States 3
0:T0.0.s=0;
0:T0.0.s=1;
0:T0.0.s=3;
Ok
Witnesses
Positive: 3 Negative: 0
Condition forall (true)
Observation while-toofar Always 3 0
Hash=18af3c2d65d7c6d9c13ed93a21870763

